component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -55,683 +55,686 @@
-		}
-		String tenantDomain = context.getTenantDomain();
-		Object getPropertiesFromIdentityConfig =
-				context.getProperty(TOTPAuthenticatorConstants.GET_PROPERTY_FROM_IDENTITY_CONFIG);
-		//If the config file is not in registry and the it is super tenant, getting the property from local.
-		// Else getting it from context.
-		if ((getPropertiesFromIdentityConfig != null ||
-		     tenantDomain.equals(TOTPAuthenticatorConstants.SUPER_TENANT_DOMAIN))) {
-			return Boolean.parseBoolean(IdentityHelperUtil.getAuthenticatorParameters(
-					context.getProperty(TOTPAuthenticatorConstants.AUTHENTICATION).toString())
-					.get(TOTPAuthenticatorConstants.ENROL_USER_IN_AUTHENTICATIONFLOW));
-		} else {
-			return Boolean.parseBoolean((context.getProperty(
-					TOTPAuthenticatorConstants.ENROL_USER_IN_AUTHENTICATIONFLOW).toString()));
-		}
-	}
+
+    private static final Log log = LogFactory.getLog(TOTPUtil.class);
+
+    /**
+     * Encrypt the given plain text.
+     *
+     * @param plainText The plaintext value to be encrypted and base64 encoded
+     * @return Base64 encoded string
+     * @throws CryptoException On error during encryption
+     */
+    public static String encrypt(String plainText) throws CryptoException {
+
+        return CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(plainText.getBytes(Charsets.UTF_8));
+    }
+
+    /**
+     * Decrypt the given cipher text.
+     *
+     * @param cipherText The string which needs to be decrypted
+     * @return Base64 decoded string
+     * @throws CryptoException On an error during decryption
+     */
+    public static String decrypt(String cipherText) throws CryptoException {
+
+        return new String(CryptoUtil.getDefaultCryptoUtil().base64DecodeAndDecrypt(cipherText), Charsets.UTF_8);
+    }
+
+    public static String getTOTPIssuerDisplayName(String tenantDomain, AuthenticationContext context)
+            throws TOTPException {
+
+        String issuer = null;
+        if (TOTPAuthenticatorConstants.SUPER_TENANT_DOMAIN.equals(tenantDomain) ||
+                Boolean.parseBoolean(getTOTPParameters().get(TOTPAuthenticatorConstants.TOTP_COMMON_ISSUER))) {
+            issuer = getTOTPParameters().get(TOTPAuthenticatorConstants.TOTP_ISSUER);
+        } else if (context == null) {
+            issuer = getIssuerFromRegistry(tenantDomain);
+        } else if (context.getProperty(TOTPAuthenticatorConstants.TOTP_ISSUER) != null) {
+            issuer = (String) context.getProperty(TOTPAuthenticatorConstants.TOTP_ISSUER);
+        }
+        if (StringUtils.isBlank(issuer)) {
+            issuer = tenantDomain;
+        }
+        return issuer;
+    }
+
+    /**
+     * Get xml file data from registry and get the value for Issuer.
+     *
+     * @param tenantDomain
+     * @return
+     * @throws TOTPException On error during passing XML content or creating document builder.
+     */
+    private static String getIssuerFromRegistry(String tenantDomain) throws TOTPException {
```

Comment:
There are duplicate codes in getIssuerFromRegistry, getEncodingMethodFromRegistry and getTimeStepSizeFromRegistry methods. Can we use a single method and parameterize the type Ex(type as issuer, encodingMethod, timeStepSize)

Reply:
Resolved with https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/77/commits/361df60ef387822f489ddafe9bccd517f32bf053.